### PR TITLE
Add postgres

### PR DESCRIPTION
**Pull request by chriswedgwood**
### What change does this PR introduce?
We want to be using a mature relational database. Up till now we have been using sqllite which is okay for the tutorial and learning but is not performant. It is also best practice to use as much of the same technology through all our environments as possible

### Notes

I have added psycopg2 and django-environ(this is to use environment variables) to the requirements file. You will need to install requirements again (pip install -r requirements.txt)

Instructions for local DB creation have been added to the readme

I have put instructions in the Readme to add a .env file for environment variables. This contains one for DATABASE_URL which is used

### Ticket

https://trello.com/c/IV6gJuLx/36-change-our-db-to-postgres-update-setup-instructions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,21 +80,28 @@ $ psql
...
 <pre>
 # CREATE DATABASE listings;
-# CREATE ROLE listings WITH LOGIN PASSWORD '<i>your password not this</i>';
+# CREATE ROLE listings WITH LOGIN PASSWORD '<i>your_password_not_this</i>';
 # GRANT ALL PRIVILEGES ON DATABASE listings TO listings;
 # ALTER USER listings CREATEDB;
 </pre>
 
+In the root of your project create a file called .env and add the following:
+
```
Comment (Icicleta) [inline README.md]: @chriswedgwood, one question please, do we need to add this file to .gitignore?

Comment (CianciuStyles) [reply README.md]: I would say yes, as usually the DATABASE_URL requires the username and password for authentication.

Comment (Icicleta) [reply README.md]: maybe he meant outside of the project?

Comment (chriswedgwood) [reply README.md]: The .env is already in .gitignore under #environments :)

Comment (Icicleta) [reply README.md]: oh yes, I see it now :) thanks @chriswedgwood

File: listings_app/settings.py [hunk 2]
```diff
@@ -11,6 +11,11 @@
 """
 
 import os
+import environ
+
+ROOT_DIR = environ.Path(__file__) - 2  # two folder back to root
```
Comment (Icicleta) [inline listings_app/settings.py]: @chriswedgwood, another question please, why root is two folder back? the settings file is already at the root of the app, isn't it? or is root something else.

Comment (chriswedgwood) [reply listings_app/settings.py]: I've added # (listings-app/listings_app/settings.py - 2 = listings-app/)
to the settings file
We need to go back to the root of the project as that's where the .env file will be

Comment (Icicleta) [reply listings_app/settings.py]: oh god, I'm stupid, I had on my mind that settings file was on listing-app folder.
However still one more thing, sorry if it's stupid question again, settings is a file, not a folder, so shouldn't be listings-app/listings_app/ -1 = listing-app? or doesn't it work differently than other paths systems?

Comment (chriswedgwood) [reply listings_app/settings.py]: No such thing as a stupid question in this project :)

The best way to check this in action is to create a python file in the same location as settings .py

Put this in it:

```
import environ

print(environ.Path(__file__)-2)
```
Then run it and fiddle with the 2 and see what comes back

File: listings_app/settings.py [hunk 3]
```diff
@@ -11,6 +11,11 @@
 """
 
 import os
+import environ
+
+ROOT_DIR = environ.Path(__file__) - 2  # two folder back to root # (listings-app/listings_app/settings.py - 2 = listings-app/)
+env = environ.Env()
+env.read_env(str(ROOT_DIR.path('.env')))
```
Comment (lili2311) [inline listings_app/settings.py]: I would prefer to use `env_utils` https://pypi.org/project/env_utils/ it is commonly used

```env_utils import (
    get_bool,
    get_env,
    get_int,
    get_list,
)``

This way we can set env like this: `export DATABASE_URL=something` and it works with Heroku as well as soon we deploy this. And to read it is is simply `DATABASE_URL = get_env('DATABASE_URL')`

The proposed approach seems to require a hard file & changing our ROOT_DIR which I feel is not needed. Was there a particular reasoning behind this approach?

**Comment by lili2311** [comment]
👋 Is this a forked PR? I can't seem to pull it down

File: listings_app/settings.py (hunk 3 above)
Comment (chriswedgwood) [reply listings_app/settings.py]: I was replicating a pattern used in django-cookiecutter.  I've not used env_utils am more than happy to use it. Let me read through its documentation and re-submit.

**Comment by chriswedgwood** [comment]
>  Is this a forked PR? I can't seem to pull it down

Yes its forked. Is that an incorrect approach?

**Comment by lili2311** [comment]
@chriswedgwood let me add you properly to this repo, it makes it easier for us all to pull things down. Do you mind using the repo directly for the next PR?

**Comment by chriswedgwood** [comment]
Im closing this and will open a new pull request against develop branch

**Comment by chriswedgwood** [comment]
> @chriswedgwood let me add you properly to this repo, it makes it easier for us all to pull things down. Do you mind using the repo directly for the next PR?

Just saw this. Are you asking don't use forked repos, just create branches within this repo?

**Pull request merged**